src/main/webapp/app/stores/curation-page.store.ts

```diff
@@ -7,20 +7,20 @@ import { alterationControllerClient } from 'app/shared/api/clients';
 
 export type MutationQuery = {
   name: string;
-  alterations: FirebaseAlt[];
+  alterations: FirebaseAlt[] | null;
 };
 
 export type AnnotatedAltsQuery = {
   query: AnnotateAlterationBody;
-  result: AlterationAnnotationStatus;
+  result: AlterationAnnotationStatus | null;
 };
 
 export type AnnotatedAltsCache = {
   loading: boolean;
-  error: Error;
+  error: Error | null;
   cache: { [key: string]: AnnotatedAltsQuery };
-  get: (hugoSymbol: string, mutations: MutationQuery[]) => AlterationAnnotationStatus[];
-  fetch: (hugoSymbol: string, mutations: MutationQuery[]) => Promise<AlterationAnnotationStatus[]>;
+  get: (hugoSymbol: string, mutations: MutationQuery[]) => (string | AlterationAnnotationStatus | null | undefined)[];
```

Comment:
```ts
export class CurationPageStore {
  public annotatedAltsCache: AnnotatedAltsCache = {
    loading: false,
    cache: {},
    error: null,
    get: (hugoSymbol: string, mutations: MutationQuery[]) => this.getAnnotatedAltsCache(hugoSymbol, mutations),
    fetch: (hugoSymbol: string, mutations: MutationQuery[]) => this.fetchAnnotatedAltsCache(hugoSymbol, mutations),
  };

  constructor(protected rootStore: IRootStore) {
    makeAutoObservable(this);
  }

  async annotateAlterations(queries: AnnotateAlterationBody[]) {
    try {
      this.annotatedAltsCache.loading = true;
      const result = await alterationControllerClient.annotateAlterations(queries);
      result.data.forEach(queryStatus => {
        if (queryStatus.queryId) {
          this.annotatedAltsCache.cache[queryStatus.queryId].result = queryStatus;
        } else {
          throw new Error('queryId is missing');
        }
      });
    } catch (responseError) {
      this.annotatedAltsCache.error = responseError as Error;
    } finally {
      this.annotatedAltsCache.loading = false;
    }
  }

  async fetchAnnotatedAltsCache(hugoSymbol: string, mutations: MutationQuery[]): Promise<(AlterationAnnotationStatus | null | undefined)[]> {
    const queries = mutations.reduce((acc, mutation) => {
      acc.push(...createMutationQuery(hugoSymbol, mutation));
      return acc;
    }, [] as AnnotateAlterationBody[]);

    const unannotatedQueries = queries.filter(query => query.queryId && !this.annotatedAltsCache.cache[query.queryId]);
    if (unannotatedQueries.length > 0) {
      unannotatedQueries.forEach(query => {
        if (query.queryId) {
          this.annotatedAltsCache.cache[query.queryId] = {
            query,
            result: null,
          };
        } else {
          throw new Error('queryId is empty');
        }
      });
      await this.annotateAlterations(unannotatedQueries);
    }
    return queries
      .filter(query => query.queryId && this.annotatedAltsCache.cache[query.queryId] && this.annotatedAltsCache.cache[query.queryId].result)
      .map(query => query.queryId && this.annotatedAltsCache.cache[query.queryId].result);
  }

  getAnnotatedAltsCache(hugoSymbol: string, mutations: MutationQuery[]) {
    const queries = mutations.reduce((acc, mutation) => {
      acc.push(...createMutationQuery(hugoSymbol, mutation));
      return acc;
    }, [] as AnnotateAlterationBody[]);
    return queries
      .filter(query => query.queryId && this.annotatedAltsCache.cache[query.queryId] && this.annotatedAltsCache.cache[query.queryId].result)
      .map(query => query.queryId && this.annotatedAltsCache.cache[query.queryId].result);
  }
}
```

These functions can return an empty string

```ts
get: (hugoSymbol: string, mutations: MutationQuery[]) => this.getAnnotatedAltsCache(hugoSymbol, mutations),
    fetch: (hugoSymbol: string, mutations: MutationQuery[]) => this.fetchAnnotatedAltsCache(hugoSymbol, mutations),
```

It's because of the `&&` condition in this map function.
```ts
    return queries
      .filter(query => query.queryId && this.annotatedAltsCache.cache[query.queryId] && this.annotatedAltsCache.cache[query.queryId].result)
      .map(query => query.queryId && this.annotatedAltsCache.cache[query.queryId].result);
```

Since empty string is falsy javascript will return an empty string.
I'll fix this.